File: 10892.cpp
#include <bits/stdc++.h>
using namespace std;

vector<long long> divisors(long long n){
  vector<long long> divs;

  for(long long i = 1 ; i <= pow(n, 0.5) ; i++){
    if(n%i == 0){
      divs.push_back(i);
      if(n/i != i) divs.push_back(n/i);
    }
  }
  return divs;
}

long long mcd(long long a, long long b){
  if(min(a, b) == 0) return max(a, b);
  else return mcd(max(a, b)%min(a, b), min(a, b));
}


int main(){
  long long n, card;

  while(1){
    cin >> n;
    if(n == 0) break;

    vector<long long> divs = divisors(n);

    card = 0;

    for(long long i = 0 ; i < divs.size() ; i++)
      for(long long j = 0 ; j < divs.size() ; j++)
        if(i != j and ((divs[i]*divs[j])/mcd(divs[i], divs[j])) == n) card++;

    cout << n << " " << card/2 + 1 << endl;

  }




  return 0;
}

File: 10892.py
import math


def getDivisors(n):
    divs = list()
    for i in range(1, int(math.pow(n, 0.5)) + 1):
        if(n%i == 0): divs.append(int(i))
        if(n%i == 0 and n/i != i): divs.append(int(n/i))
    return divs

def mcd(a, b):
    if(min(a, b) == 0): return max(a, b)
    else: return mcd(max(a, b)%min(a, b), min(a, b))

def main():

    while(1):
        n = int(input())
        if(n == 0): break

        divisors = getDivisors(n)


        card = 1
        for i in range(len(divisors)):
            for j in range(len(divisors)):
                if(i != j and ((divisors[i]*divisors[j])/mcd(divisors[i], divisors[j])) == n): card += 1

        card = math.ceil(card/2)
        print(n, card)


main()
